runelite-client/src/main/java/net/runelite/client/plugins/puzzlesolver/SecurityStrongholdAnswer.java

```diff
@@ -0,0 +1,103 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.puzzlesolver;
+
+import java.util.ArrayList;
+import java.util.List;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import net.runelite.api.widgets.Widget;
+
+@Getter
+@RequiredArgsConstructor
+enum SecurityStrongholdAnswer
+{
+	QUESTION_1("Secure my device and reset my RuneScape password."),
+	QUESTION_2("Don't share your information and report the player."),
+	QUESTION_3("Authenticator and two-step login on my registered email."),
+	QUESTION_4("Nobody."),
+	QUESTION_5("Don't tell them anything and click the 'Report Abuse' button."),
+	QUESTION_6("Talk to any banker in RuneScape."),
+	QUESTION_7("Don't give them my password."),
+	QUESTION_8("Don't give out your password to anyone. Not even close friends."),
+	QUESTION_9("Do not visit the website and report the player who messaged you."),
+	QUESTION_10("Inform Jagex by emailing reportphishing@jagex.com."),
+	QUESTION_11("Don't give out your password to anyone. Not even close friends."),
+	QUESTION_12("Read the text and follow the advice given."),
+	QUESTION_13("Only on the RuneScape website."),
+	QUESTION_14("Through account settings on runescape.com."),
+	QUESTION_15("Set up 2 step authentication with my email provider."),
+	QUESTION_16("No way! You'll just take my gold for your own! Reported!."),
+	QUESTION_17("Secure my device and reset my RuneScape password."),
+	QUESTION_18("Don't type in my password backwards and report the player."),
+	QUESTION_19("Report the stream as a scam. Real Jagex streams have a 'verified' mark."),
+	QUESTION_20("Use the Account Recovery System."),
+	QUESTION_21("Report the incident and do not click any links."),
+	QUESTION_22("Don't click any links, forward the email to reportphishing@jagex.com."),
+	QUESTION_23("Me."),
+	QUESTION_24("The birthday of a famous person or event."),
+	QUESTION_25("No, you should never allow anyone to level your account."),
+	QUESTION_26("Only on the RuneScape website."),
+	QUESTION_27("Virus scan my device then change my password."),
+	QUESTION_28("No, you should never buy a RuneScape account."),
+	QUESTION_29("No way! I'm reporting you to Jagex!"),
+	QUESTION_30("No way! You'll just take my gold for your own! Reported!"),
+	QUESTION_31("Politely tell them no and then use the 'Report Abuse' button."),
+	QUESTION_32("No."),
+	QUESTION_33("Report the player for phishing."),
+	QUESTION_34("Don't give them the information and send an 'Abuse Report'.");
+
+	private static final List<String> MATCHES = new ArrayList<>();
```

Comment:
Would it be better to use a HashSet instead of a list? Lookup would be O(1) instead of O(n), correct me if I'm wrong tho.

Reply:
Are you sure you don't work at my office? We were having this exact discussion last week. You stalker.

HashSet is generally viewed as being `O(1)` (And in this case will be) but can, in very edge cases, get performance as bad as `log(n)` (But how often do hash collisions actually occur?)

An ArrayList is indeed `O(n)` since it just uses `.indexOf()`

With such a tiny collection we're into tiny amounts of nano-second difference between the two.

So tiny difference, but given the use-case, no real reason to NOT use the set over the list.

Reply:
Yep that's completely understandable, for this particular use case it doesn't really matter in terms of optimization; however, when a newer person is looking at your code in the future for example, it may be best to write it in the best way in case that user does copy pasta but for a larger data set or is looking at your code as a learning opportunity. So I'd still stand by switching it over to a HashSet, but it's understandable if you don't switch it because in terms of optimization it barely contributes.

Reply:
I will change it to HashSet, the order isn't important in this case anyway.